js/app.js

```diff
@@ -11,11 +11,127 @@ Calculator.prototype.getHistoryAsString = function() {
+//     // 4. dodaj do historii operacji to działanie w fomie: 1 + 1 = 2
+
+//     const number1 = parseFloat(num1);
+//     const number2 = parseFloat(num2);
+
+//     if (isNaN(number1) || isNaN(number2)) {
+//         alert('Podana wartość musi być liczbą! Wprowadź poprawne dane!');
+//         location.reload();
+//     }
+
+//     const result = number1 + number2;
+//     const operation = `${number1} + ${number2} = ${result}`;
+//     this.history.push(operation);
+// }
+
+// Calculator.prototype.subtract = function(num1, num2) {
+
+//     const number1 = parseFloat(num1);
+//     const number2 = parseFloat(num2);
+
+//     if (isNaN(number1) || isNaN(number2)) {
+//         alert('Podana wartość musi być liczbą! Wprowadź poprawne dane!');
+//         location.reload();
+//     }
+
+//     const result = number1 - number2;
+//     const operation = `${number1} - ${number2} = ${result}`;
+//     this.history.push(operation);
+// }
+
+// Calculator.prototype.multiply = function(num1, num2) {
+
+//     const number1 = parseFloat(num1);
+//     const number2 = parseFloat(num2);
+
+//     if (isNaN(number1) || isNaN(number2)) {
+//         alert('Podana wartość musi być liczbą! Wprowadź poprawne dane!');
+//         location.reload();
+//     }
+
+//     const result = number1 * number2;
+//     const operation = `${number1} * ${number2} = ${result}`;
+//     this.history.push(operation);
+// }
+
+// Calculator.prototype.divide = function(num1, num2) {
+
+//     const number1 = parseFloat(num1);
+//     const number2 = parseFloat(num2);
+
+//     if (isNaN(number1) || isNaN(number2)) {
+//         alert('Podana wartość musi być liczbą! Wprowadź poprawne dane!');
+//         location.reload();
+//     }
+
+//     const result = number1 / number2;
+//     const operation = `${number1} / ${number2} = ${result}`;
+//     this.history.push(operation);
+// }
+
+// Calculator.prototype.exponentiation = function(num1, num2) {
+
+//     const number1 = parseFloat(num1);
+//     const number2 = parseFloat(num2);
+    
+
+//     if (isNaN(number1) || isNaN(number2)) {
+//         alert('Podana wartość musi być liczbą! Wprowadź poprawne dane!');
+//         location.reload();
+//     }
+
+//     let result = 1;
+//     for (let i = 0; i < number2; i++){
+//         result *=number1;
+//     }
+
+//     const operation = `${number1} ^ ${number2} = ${result}`;
+//     this.history.push(operation);
+// }
+
+Calculator.prototype.doOperation = function(operator, num1, num2){
+    const number1 = parseFloat(num1);
+    const number2 = parseFloat(num2);
+    let result;
+    
+
+    if (isNaN(number1) || isNaN(number2)) {
+        alert('Podana wartość musi być liczbą! Wprowadź poprawne dane!');
+        location.reload();
+    }
+
+    switch(operator){
+        case '+':
+            result = number1 + number2;
+            break;
+
+        case '-':
+            result = number1 - number2;
+            break;
```

Comment:
Podział na funkcje sam w sobie nie był taki zły ponieważ mają 100 różnych operacji, funkcja `doOperation` będzie bardzo długa.
Spróbuj zostać przy osobnych funkcjach, ale spróbuj "dynamicznie" uruchamiać funkcje.
Możesz się wzorować na tym artykule: https://devmentor.pl/b/mniej-instrukcji-warunkowych

PS. `switch` to pewnego rodzaju `if`, który ładniej wygląda więc nie jest to do końca rozwiązanie problemu :)